app/src/main/java/ro/code4/deurgenta/helper/MapViewUtils.kt

```diff
@@ -0,0 +1,380 @@
+
+    /**
+     * Location client used to localize the user.
+     */
+    private var fusedLocationClient: FusedLocationProviderClient
+
+    /**
+     * Last query string.
+     */
+    private var queryString = ""
+
+    init {
+        locationRequest.priority = LocationRequest.PRIORITY_HIGH_ACCURACY;
+        locationRequest.interval = 10 * 1000; // 10 seconds
+        locationRequest.fastestInterval = 5 * 1000; // 5 seconds
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(activity)
+
+        val builder = LocationSettingsRequest.Builder()
+        builder.addLocationRequest(locationRequest)
+        locationSettingsRequest = builder.build()
+
+        settingsClient = LocationServices.getSettingsClient(activity)
+    }
+
+    fun loadMapSceneForLocation(location: Location) {
+        geoCoordinates = GeoCoordinates(location.latitude, location.longitude)
+        loadMapSceneForGeoCoordinates(geoCoordinates)
+    }
+
+    private fun loadMapSceneForGeoCoordinates(geoCoordinates: GeoCoordinates?) {
+        if (geoCoordinates == null) {
+            callback.onError(MapDataError(R.string.unknown_error, MapErrorType.UNKNOWN))
+            return
+        }
+
+        mapView.mapScene
+            .loadScene(MapScheme.NORMAL_DAY) { errorCode ->
+                if (errorCode == null) {
+
+                    val distanceInMeters = DEFAULT_DISTANCE_IN_METERS.toDouble()
+                    mapView.camera.lookAt(geoCoordinates, distanceInMeters)
+                    callback.onLoaded()
+                } else {
+                    Log.e(TAG, "error loading results:$errorCode")
+                    if (errorCode != MapError.OPERATION_IN_PROGRESS) {
+                        callback.onError(
+                            MapDataError(
+                                R.string.operation_in_progress,
+                                MapErrorType.NOT_INITIALIZED
+                            )
+                        )
+                    }
+                }
+            }
+    }
+
+
+    private fun createAndAddMarker(searchResult: Place) {
+        clearMap()
+
+        if (searchResult.geoCoordinates == null) {
+            Log.d(TAG, "no coordinates to add")
+            return
+        }
+
+        val mapMarker = createPoiMapMarker(searchResult.geoCoordinates!!)
+        mapView.mapScene.addMapMarker(mapMarker)
+
+        val mapAddress = MapAddress(
+            latitude = searchResult.geoCoordinates!!.latitude,
+            longitude = searchResult.geoCoordinates!!.longitude,
+            fullAddress = searchResult.address.addressText
+        )
+        mapAddress.streetAddress = getStreetAddress(searchResult.address)
+
+        mapMarkerList[mapAddress] = mapMarker
+    }
+
+    private fun getStreetAddress(address: Address): String {
+        return formatAddress(address)
+    }
+
+    private fun getMapViewGeoCircle(): GeoCircle {
+        return GeoCircle(geoCoordinates, 50000.0)
```

Comment:
could you extract the radius value in the config files so it could be easily changed?